scripts/generate_dump

```diff
@@ -461,7 +461,7 @@ main() {
     $MKDIR $V -p $LOGDIR
     $LN $V -s /etc $TARDIR/etc
 
-    ($TAR $V -rhf $TARFILE -C $DUMPDIR --mode=+rw \
+    ($TAR $V -rf $TARFILE -C $DUMPDIR --mode=+rw \
```

Comment:
why remove -h?

Reply:
We see below error always, for a missing reference, also most of the rc.d directory is symlink to init.d and "-h" dos hard copy of all sym links.
 
tar: sonic_dump_str-marvell-acs-1_20201105_095941/etc/rc0.d/K01docker: File removed before we read it
tar: sonic_dump_str-marvell-acs-1_20201105_095941/etc/rc1.d/K01docker: File removed before we read it
tar: sonic_dump_str-marvell-acs-1_20201105_095941/etc/rc2.d/S01docker: File removed before we read it
tar: sonic_dump_str-marvell-acs-1_20201105_095941/etc/rc3.d/S01docker: File removed before we read it
tar: sonic_dump_str-marvell-acs-1_20201105_095941/etc/rc4.d/S01docker: File removed before we read it
tar: sonic_dump_str-marvell-acs-1_20201105_095941/etc/rc5.d/S01docker: File removed before we read it
tar: sonic_dump_str-marvell-acs-1_20201105_095941/etc/rc6.d/K01docker: File removed before we read it
Tar append operation failed. Aborting for safety.

# ls -l /etc/rc1.d/K01docker
lrwxrwxrwx 1 root root 16 Oct 31 04:14 /etc/rc1.d/K01docker -> ../init.d/docker

# ls -l /etc/init.d/docker
ls: cannot access '/etc/init.d/docker': No such file or directory

Reply:
why is ../init.d/docker missing here?

Reply:
i will resolve the init.d/docker missing issue

Reply:
what safety is here? i still think it is correct to do the dereference here since the file maybe in different directory and -h allow us to capture the file.

Reply:
added -h option back.
we see same issue in intel platform as well. /etc/init.d/docker is missing.
Do you have any suggestion to fix this?

**logs:**

root@sonic:/home/admin# generate_dump
tar: sonic_dump_sonic_20190312_180102/etc/rc0.d/K01docker: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/rc1.d/K01docker: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/rc2.d/S01docker: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/rc3.d/S01docker: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/rc4.d/S01docker: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/rc5.d/S01docker: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/rc6.d/K01docker: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/fancontrol.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/platform-modules-dx010.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/platform-modules-haliburton.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/platform-modules-s5232f.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/platform-modules-s5248f.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/platform-modules-s6000.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/platform-modules-s6100.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/platform-modules-seastone2.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/platform-modules-silverstone.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/platform-modules-z9100.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/platform-modules-z9264f.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/platform-modules-z9332f.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/qfx5200-platform-init.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/qfx5210-platform-init.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/s6100-lpc-monitor.service: File removed before we read it
tar: sonic_dump_sonic_20190312_180102/etc/systemd/system/multi-user.target.wants/z9100-lpc-monitor.service: File removed before we read it
Tar append operation failed. Aborting for safety.
/usr/local/bin/generate_dump: line 521: /usr/bin/hw-management-generate-dump.sh: No such file or directory
cp: cannot stat '/tmp/hw-mgmt-dump*': No such file or directory
tar: sonic_dump_sonic_20190312_180102/hw-mgmt/hw-mgmt-dump*: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
tar append operation failed. Aborting to prevent data loss.
/var/dump/sonic_dump_sonic_20190312_180102.tar.gz